Sources/MuxPlayerSwift/PublicAPI/FairPlay/MuxFairPlayContentKeySessionDelegate.swift

```diff
@@ -0,0 +1,107 @@
+///   - ``AVFoundation/AVPlayerLayer/init(playbackID:playbackOptions:)``
+public class MuxFairPlayContentKeySessionDelegate: NSObject, AVContentKeySessionDelegate {
+    
+    // We wrap our internal delegate in a facade so we don't have to expose the internal plumbing in our
+    // ContentKeySessionDelegate, which mostly exists for testability and isn't useful to customers
+    private let internalDelegate: ContentKeySessionDelegate< DefaultFairPlayStreamingSessionManager<AVContentKeySession>
+    >?
+    private let sessionManager: DefaultFairPlayStreamingSessionManager<AVContentKeySession>
+    
+    public func contentKeySession(
+        _ session: AVContentKeySession,
+        didProvide keyRequest: AVContentKeyRequest
+    ) {
+        internalDelegate?.contentKeySession(session, didProvide: keyRequest)
+    }
+    
+    public func contentKeySession(
+        _ session: AVContentKeySession,
+        didProvideRenewingContentKeyRequest keyRequest: AVContentKeyRequest
+    ) {
+        internalDelegate?.contentKeySession(session, didProvideRenewingContentKeyRequest: keyRequest)
+    }
+    
+    public func contentKeySession(
+        _ session: AVContentKeySession,
+        contentKeyRequestDidSucceed keyRequest: AVContentKeyRequest
+    ) {
+        internalDelegate?.contentKeySession(session, contentKeyRequestDidSucceed: keyRequest)
+    }
+    
+    public func contentKeySession(
+        _ session: AVContentKeySession, contentKeyRequest
+        keyRequest: AVContentKeyRequest, didFailWithError
+        err: any Error
+    ) {
+        internalDelegate?.contentKeySession(session, contentKeyRequest: keyRequest, didFailWithError: err)
+    }
+    
+    public func contentKeySession(
+        _ session: AVContentKeySession,
+        shouldRetry keyRequest: AVContentKeyRequest,
+        reason retryReason: AVContentKeyRequest.RetryReason
+    ) -> Bool {
+        if let internalDelegate = self.internalDelegate {
+            return internalDelegate.contentKeySession(
+                session, shouldRetry: keyRequest, reason: retryReason
+            )
+        } else {
+            return false
+        }
+    }
+    
+    /// Make a new instance for the given Mux PlaybackID, using the given playback and DRM tokens
+    /// - SeeAlso: https://www.mux.com/docs/guides/mux-player-ios#secure-your-playback-experience
+    public init(playbackID: String, playbackToken: String, drmToken: String) {
+        #if targetEnvironment(simulator)
+        // Creating an AVContentKeySession for fairplay will cause a silent crash on simulators, so
+        //  we stub out functionality using a .clearKey session and dummy session delegate
+        self.sessionManager = DefaultFairPlayStreamingSessionManager(
+            contentKeySession: AVContentKeySession(keySystem: .clearKey),
+            urlSession: .shared,
+            errorDispatcher: PlayerSDK.shared.monitor
+        )
+        self.internalDelegate = nil
+        #else
```

Comment:
Just a thought, how about we wrap the `internalDelegate` itself and all calls to it with `#if !targetEnvironment(simulator)`? Currently the optional is effectively a platform difference instead of a runtime difference.